### DISCUSSION TITLE
DPMS on wayland ?

### DISCUSSION THREAD

**Discussion by MrDisguised** [hyprwm/Hyprland#1974] [Q&A] [answered]
In xorg we can use` xfce-power-manager` to manage DPMS events like suspending the computer / hibernating the computer after specified time interval. How can I do that in wayland ? I read the wiki and there are no instructions in the wiki on how to set it up (there are options to turn the dpms of the monitor on/off but there are no instructions on how to set DPMS timers). Can anyone help me ?

**Comment by grappas** [comment]
Swayidle

**Reply by MrDisguised** [reply]
> Swayidle

I tried swayidle and swaylock but the problem is when I command swayilde to lock my screen using swaylock it works but then when I try to type my password to unlock my computer swaylock doesn't respond to my keyboard / mouse or any other input devices. Please help :'(

**Reply by grappas** [reply]
hyprland.conf
```
misc {
        mouse_move_enables_dpms = true
        key_press_enables_dpms = true
}
```
Otherwise you have to bind dpms enable.
Here's mine swayidle.sh (no swaylock though):
```
#!/bin/sh
swayidle -d -w \
                timeout 240 'hyprctl dispatch dpms off' \
                resume 'hyprctl dispatch dpms on' &
```

**Reply by MrDisguised** [reply]
The bash script you gave didn't work. After executing the script my computers turns it screen off after 240 seconds and shows a blue light (which is all right) but after another 240 seconds my computer screen shows no signal and turns off again ( now it shows a red light ) . I cannot turn on my monitor again ( by pressing keyboard or clicking on mouse buttons) . I have to press the power button on my cpu cabinet 2 times which shuts down the computer and again reboots it. I don't want that.

**Comment by grappas** [comment]
> no swaylock though

Just add swaylock.
> misc {
        mouse_move_enables_dpms = true
        key_press_enables_dpms = true
}

Add it to misc section in hyprland.conf

**Reply by MrDisguised** [reply]
> > no swaylock though
> > Just add swaylock.
> > misc {
> > mouse_move_enables_dpms = true
> > key_press_enables_dpms = true
> > }
> > Add it to misc section in hyprland.conf

already did it. I read the wiki

**Reply by MrDisguised** [reply]
> > no swaylock though
> 
> Just add swaylock.
> 
> > misc {
> > mouse_move_enables_dpms = true
> > key_press_enables_dpms = true
> > }
> 
> Add it to misc section in hyprland.conf

Here is my hyprland config file - https://paste.rs/WDk

**Reply by grappas** [reply, marked as answer]
> exec-once = swayidle timeout 600 "swaylock"

exec-once = swayidle timeout 600 'hyprctl dispatcher dpms off && swaylock'

**Comment by MrDisguised** [comment]
> > exec-once = swayidle timeout 600 "swaylock"
> 
> exec-once = swayidle timeout 600 'hyprctl dispatcher dpms off && swaylock'

Thanks a lot!! That command worked. Btw can I create a script which will lock my screen and turn off my display after 600 seconds , it will suspend my session after another 300 seconds and after another 300 seconds it will hibernate my pc ? Like this ? - 

![image](https://user-images.githubusercontent.com/128170542/230721090-9ad2a369-3acf-47df-a471-1cc0d5770210.png)

**Reply by grappas** [reply]
try it with lower timeout yourself 😉
![obraz](https://user-images.githubusercontent.com/1036957/230721771-825fe0d1-fbed-4fb4-a2de-24e8613be138.png)

**Reply by MrDisguised** [reply]
> try it with lower timeout yourself wink ![obraz](https://user-images.githubusercontent.com/1036957/230721771-825fe0d1-fbed-4fb4-a2de-24e8613be138.png)

Thanks a lot for helping me :pray: :pray: :D
